### PR TITLE
enable openvr plugin on linux and update OpenVR to 1.0.10

### PR DESCRIPTION
**Pull request by ChristophHaag**
also disable threaded submit on non windows platform (very slow and crashes steamvr compositor on archlinux)
TODO: non windows implementation of getPreferredAudioOutDevice and getPreferredAudioOutDevice

I have been testing this on Archlinux and since it seems to be fully working (after manually choosing audio input and output in the settings) I thought I'd make it a pull request.

Issues:
As mentioned the preferred audio devices are only chosen on windows right now. I could probably try an implementation before merging.
For linking with the OpenVR plugin I had to add -fpic to glew's c flags. Not sure if it's good like that on windows?
I'm not sure if QCoreApplication miniApp(__argc, __argv); in OpenVrHelpers.cpp can do anything with the args but the current solution is a windows-only thing, so I just made them to 0 on non windows platforms and it works.

If you want to actually try it on Linux, you may want to look at my master branch that is currently on RELEASE-7071 (to connect to the stable domains) and has a couple other commits that I needed to get it to work on radeonsi: https://github.com/ChristophHaag/hifi/commits/master

Related issue: https://github.com/highfidelity/hifi/issues/10098

### CODE REVIEW

**Comment by cozza13** [comment]
OK to test

**Comment by ChristophHaag** [comment]
fixed all the warnings, that should fix the linux and windows build. not sure what about the undefined symbols on mac os though, it sounds like that's a problem in the upstream distributed openvr library.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3242/HighFidelity-Beta-PR11318-8ec38979086f0a67bcbcd9ecf7ced61cac788ea6.exe

**Comment by ZappoMan** [comment]
gustavo test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3465/HighFidelity-Beta-PR11318-8ec38979086f0a67bcbcd9ecf7ced61cac788ea6.exe

**Comment by themelissabrown** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3895/HighFidelity-Beta-PR11318-8ec38979086f0a67bcbcd9ecf7ced61cac788ea6.exe

**Comment by ChristophHaag** [comment]
At https://github.com/ChristophHaag/hifi I rebased all my stuff on RELEASE-7262 and it turns out draco also needs to be built with -fPIC to link the openvr plugin https://github.com/ChristophHaag/hifi/commit/cf9595055e7103ac26ba7b9877203f065466e253 and I had to disable some asserts.

Anyway, as I said I don't know what's up with the build failure on Mac OS X. Someone should probably investigate this and I strongly suspect someone needs to talk to Valve about fixing their libopenvr_api library for Mac OS X.

Other than that it requires stable SteamVR. With SteamVR beta High Fidelity renders wrong for which I have made those bug reports:
https://github.com/highfidelity/hifi/issues/11438
https://github.com/ValveSoftware/SteamVR-for-Linux/issues/72

And so I don't forget: Apparently I'm missing advanced motion controller settings in the menu so I can't set up jumping in VR. Not sure what's up with that.

Long story short, it would be nice if someone from hifi who uses Linux could check this out and make sure it's running well.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4077/HighFidelity-Beta-PR11318-8ec38979086f0a67bcbcd9ecf7ced61cac788ea6.exe

**Comment by ChristophHaag** [comment]
Not sure why github shows several commits here. I did `git reset --hard RELEASE-7360` and cherry picked the openvr commit from my master branch.

In an unrelated change 4d2f16efc3f1d37521f761c30d4c513a4b9f06b1 added `-DCMAKE_POSITION_INDEPENDENT_CODE=ON` to a couple of external libraries so that's cool. In this commit I only have to add it to draco now. That's probably much more portable to other compilers than the -fPIC flag.

The rendering issue with SteamVR Beta is gone now.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4541/HighFidelity-Beta-PR11318-8d73218eb405ede788024341bdb4fb46585ef5ed.exe

**Comment by ChristophHaag** [comment]
Next attempt: Since there's a build failure on mac os x and I don't actually want to pack enabling linux support together with updating to OpenVR 1.0.10 into the same commit and I think openvr didn't have mac os x support with the current OpenVR 1.0.6 anyway, I think I should limit it to linux first. The Mac OS X build can be fixed later by someone who actually has Mac OS X and can test locally what goes wrong.

I don't like using `if (WIN32 OR (UNIX AND NOT ANDROID AND NOT APPLE))` for testing for desktop linux in cmake, but I haven't seen something like if(LINUX). Maybe it should be introduced.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4547/HighFidelity-Beta-PR11318-a9364a9293a50d2086f7ad6c1c71f5200ce6d88c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4547/HighFidelity-Beta-PR11318-a9364a9293a50d2086f7ad6c1c71f5200ce6d88c.exe

**Comment by ZappoMan** [comment]
@ChristophHaag - Can you try to rebase this or update it to the latest master?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5061/HighFidelity-Beta-PR11318-377baaf043fb658517584385c02c0624489a2ed4.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5061/HighFidelity-Beta-PR11318-377baaf043fb658517584385c02c0624489a2ed4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5359/HighFidelity-Beta-PR11318-38f01ff4ab273e56cacc6f88901796edd70c1530.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5359/HighFidelity-Beta-PR11318-38f01ff4ab273e56cacc6f88901796edd70c1530.dmg

**Comment by ChristophHaag** [comment]
Someone informed me that Valve has to versions of the openvr dylib for mac os x in their repository. The one in lib/ is old, the one in bin/ is new. So linking this one seems to work. I have no Mac OS X here, so the Mac build seems to succeed but whether it actually works with VR, I have no idea.

As a reminder, it doesn't really run on Linux without several additional fixes/workarounds, which I have in my master branch, currently on top of RELEASE-7474 which can connect to current non-dev domains: https://github.com/ChristophHaag/hifi/. The same is likely true for Mac OS X.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5361/HighFidelity-Beta-PR11318-e33ca0e4668f625b3f81cfc2e9b33b7d39e471e5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5361/HighFidelity-Beta-PR11318-e33ca0e4668f625b3f81cfc2e9b33b7d39e471e5.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6848/app_11318_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6848/HighFidelity-Beta-PR11318-1942c22b9cb4d14c1a28faf6d4dfcb9ad74bb96b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6848/HighFidelity-Beta-PR11318-1942c22b9cb4d14c1a28faf6d4dfcb9ad74bb96b.dmg

**Comment by samcake** [comment]
THis hasn;t been touched in month,
i m going to close that until you tell me we need it.
thanks

**Comment by ChristophHaag** [comment]
Yea sure, it doesn't run in VR on Linux with just only this PR anyway, it just crashes because of unrelated things.

tl;dr of this PR thread:

For anyone who wants to run High Fidelity in VR on Linux, I just keep "maintaining" my fork at https://github.com/ChristophHaag/hifi.
If anyone from the hifi developers actually wants to support a VR enabled version on Linux, it will always be there as a reference.

"Promotional" screenshot:
screenshot_20180412_005624

There is a good chance my fork can run with SteamVR on Mac OS X too, but nobody has tested (at least nobody has reported back).
